ethergo/submitter/submitter.go

```diff
@@ -346,119 +346,211 @@ func (t *txSubmitterImpl) txTypeForChain(chainID *big.Int) (txType uint8) {
+	if useDynamic {
+		suggestedGasFeeCap, err := client.SuggestGasPrice(ctx)
+		if err != nil {
+			return fmt.Errorf("could not get gas fee cap: %w", err)
+		}
+		transactor.GasFeeCap = maxOfBig(transactor.GasFeeCap, suggestedGasFeeCap)
+		suggestedGasTipCap, err := client.SuggestGasTipCap(ctx)
 		if err != nil {
 			return fmt.Errorf("could not get gas tip cap: %w", err)
 		}
+		transactor.GasTipCap = maxOfBig(transactor.GasTipCap, suggestedGasTipCap)
+		span.SetAttributes(
+			attribute.String("suggested_gas_fee_cap", bigPtrToString(suggestedGasFeeCap)),
+			attribute.String("suggested_gas_tip_cap", bigPtrToString(suggestedGasTipCap)),
+			attribute.String("gas_fee_cap", bigPtrToString(transactor.GasFeeCap)),
+			attribute.String("gas_tip_cap", bigPtrToString(transactor.GasTipCap)),
+		)
 	} else {
-		transactor.GasPrice, err = client.SuggestGasPrice(ctx)
+		suggestedGasPrice, err := client.SuggestGasPrice(ctx)
 		if err != nil {
 			return fmt.Errorf("could not get gas price: %w", err)
 		}
+		transactor.GasPrice = maxOfBig(transactor.GasPrice, suggestedGasPrice)
+		span.SetAttributes(
+			attribute.String("suggested_gas_price", bigPtrToString(suggestedGasPrice)),
+			attribute.String("gas_price", bigPtrToString(transactor.GasPrice)),
+		)
 	}
-	t.applyBaseGasPrice(transactor, chainID)
+	return nil
+}
 
-	//nolint: nestif
-	if prevTx != nil && shouldBump {
-		gasBlock, err := t.getGasBlock(ctx, client, chainID)
-		if err != nil {
-			span.AddEvent("could not get gas block", trace.WithAttributes(attribute.String("error", err.Error())))
-			return err
-		}
+// applyGasCeil evaluates current gas values versus the configured maximum, and
+// returns an error if they exceed the maximum.
+func (t *txSubmitterImpl) applyGasCeil(ctx context.Context, transactor *bind.TransactOpts, chainID int, useDynamic bool) (err error) {
+	_, span := t.metrics.Tracer().Start(ctx, "submitter.applyGasCeil")
 
-		// if the prev tx was greater than this one, we should bump the gas price from that point
-		if prevTx.Type() == types.LegacyTxType {
-			if prevTx.GasPrice().Cmp(transactor.GasPrice) > 0 {
-				transactor.GasPrice = core.CopyBigInt(prevTx.GasPrice())
-			}
-		} else {
-			if prevTx.GasTipCap().Cmp(transactor.GasTipCap) > 0 {
-				transactor.GasTipCap = core.CopyBigInt(prevTx.GasTipCap())
-			}
+	maxPrice := t.config.GetMaxGasPrice(chainID)
 
-			if prevTx.GasFeeCap().Cmp(transactor.GasFeeCap) > 0 {
-				transactor.GasFeeCap = core.CopyBigInt(prevTx.GasFeeCap())
-			}
+	defer func() {
+		span.SetAttributes(attribute.String("max_price", bigPtrToString(maxPrice)))
+		metrics.EndSpanWithErr(span, err)
+	}()
+
+	if useDynamic {
+		if transactor.GasFeeCap.Cmp(maxPrice) > 0 {
+			return fmt.Errorf("gas fee cap %s exceeds max price %s", transactor.GasFeeCap, maxPrice)
 		}
-		gas.BumpGasFees(transactor, t.config.GetGasBumpPercentage(chainID), gasBlock.BaseFee, maxPrice)
+		// TODO: should maxPrice apply to GasTipCap as well?
```

Comment:
We just need to enforce the invariant that `GasTipCap <= GasFeeCap`, so that this TODO check could be avoided

Reply:
Done here: https://github.com/synapsecns/sanguine/pull/2462/commits/b4394d1df4d201917205b8bd64b9beb7ea127ec6